From a7ba90c7e94e91785c55efa13e2efa1161cf4375 Mon Sep 17 00:00:00 2001
From: Gergely Nagy <algernon@madhouse-project.org>
Date: Sat, 30 Jun 2012 13:17:01 +0200
Subject: [PATCH] Symbol versioning

Patch by Alastair McKinstry <mckinstry@debian.org> 22 Jun 2005
Later edited by Debian package maintainer.
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=315419
---
 src/Makefile.am   |    2 +-
 src/aalib.1.4.ver |    6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)
 create mode 100644 src/aalib.1.4.ver

diff --git a/src/Makefile.am b/src/Makefile.am
index ad23666..b62d4d8 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -57,7 +57,7 @@ libaa_la_SOURCES= \
        aaattrs.c \
        aaputpixel.c \
        aarecfunc.c 
-libaa_la_LDFLAGS=@LDFLAGS@ -version-info 1:4:0
+libaa_la_LDFLAGS=@LDFLAGS@ -version-info 1:4:0 -Wl,--version-script=aalib.1.4.ver
 bin_PROGRAMS=aainfo aatest aafire aasavefont
 
 aainfo_LDADD= libaa.la
diff --git a/src/aalib.1.4.ver b/src/aalib.1.4.ver
new file mode 100644
index 0000000..fd62716
--- /dev/null
+++ b/src/aalib.1.4.ver
@@ -0,0 +1,6 @@
+AA_1.4 {
+	global:
+		aa_*; save_d; mem_d;
+	local:
+		*;
+};
-- 
1.7.10.4

